src/Components/NewAddressDialog/NewAddressDialog.js

```diff
@@ -0,0 +1,52 @@
+import './NewAddressDialog.css'
+
+export const NewAddressDialog = props => {
+	const {
+		showModal,
+		setShowModal,
+		handleChange,
+		handleSubmit,
+		addressData,
+	} = props
+
+	if (!showModal) {
+		return null
+	}
+
+	return (
+		<div
+			className='new-address-dialog'
+			onClick={() => setShowModal(false)}>
+			<form
+				onSubmit={handleSubmit}
+				className='address-form'
+				onClick={e => e.stopPropagation()}>
```

Comment:
**Suggestions:** Instead of using `e.stopPropagation()` you can use e.preventDefault() inside handleSubmit method.

Reply:
Since this a modal while clicking inside the form its event was propagating to the outer div which as setting the setShowModal to false so hence I added this. Btw theres already a e.preventDefault handleSubmit

Reply:
Thanks for pointing this out, had to rethink about why I added